### PR TITLE
Introduce gosec for security checks

### PR DESCRIPTION
**Pull request by JAORMX**
gosec does static code analysis and checks for common security issues in
golang codebases.

This PR introduces the tool, and sets it up as a test target in the Makefile;
It also ignores a false positive warning that was found.

### CODE REVIEW

File: tools/import-verifier/import-verifier.go [hunk 1]
```diff
@@ -319,6 +319,7 @@ func loadImportRestrictions(configFile string) ([]ImportRestriction, error) {
 func resolvePackage(targetPackage string) ([]Package, error) {
 	cmd := "go"
 	args := []string{"list", "-json", targetPackage}
+	// #nosec
```
Comment (stlaz) [inline tools/import-verifier/import-verifier.go]: Please add an explanation about why you are turning off the check here

Comment (JAORMX) [reply tools/import-verifier/import-verifier.go]: Will do; should I do it as part of the PR description or a comment in the code?

Comment (JAORMX) [reply tools/import-verifier/import-verifier.go]: done!

**Comment by enj** [comment]
Not opposed to this but we want this kind of stuff in the library-go Makefile thing and we need to move this repo to using that.

**Pull request merged**